auth.py

```diff
@@ -0,0 +1,236 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+from abc import ABCMeta, abstractmethod, abstractproperty
+from functools import wraps
+import hashlib
+import os.path
+import json
+import re
+
+from flask import request, Response
+
+WOTT_CREDENTIALS_PATH = '/opt/wott/credentials'
+WOTT_SCREENLY_CREDENTIAL_NAME = 'screenly_credentials'
+
+
+class Auth(object):
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def authenticate(self):
+        """
+        Let the user authenticate himself.
+        :return: a Response which initiates authentication.
+        """
+        pass
+
+    @abstractproperty
+    def is_authorized(self):
+        """
+        See if the user is authorized for the request.
+        :return: bool
+        """
+        pass
+
+    def authorize(self):
+        """
+        If the request is not authorized, let the user authenticate himself.
+        :return: a Response which initiates authentication or None if authorized.
+        """
+        if not self.is_authorized:
+            return self.authenticate()
+
+    def update_settings(self, current_password):
+        pass
+
+    @property
+    def template(self):
+        pass
+
+
+class NoAuth(Auth):
+    name = 'Disabled'
+    id = ''
+    config = {}
+
+    def is_authorized(self):
+        return True
+
+    def authenticate(self):
+        pass
+
+
+class BasicAuth(Auth):
+    name = 'Basic'
+    id = 'auth_basic'
+    config = {
+        'auth_basic': {
+            'user': '',
+            'password': ''
+        }
+    }
+
+    def __init__(self, settings):
+        self.settings = settings
+
+    def _check(self, username, password):
+        """
+        Check username/password combo against database.
+        :param username: str
+        :param password: str
+        :return: True if the check passes.
+        """
+        return self.settings['user'] == username and self.check_password(password)
+
+    def check_password(self, password):
+        hashed_password = hashlib.sha256(password).hexdigest()
+        return self.settings['password'] == hashed_password
+
+    @property
+    def is_authorized(self):
+        auth = request.authorization
+        return auth and self._check(auth.username, auth.password)
+
+    @property
+    def template(self):
+        return 'auth_basic.html', {'user': self.settings['user']}
+
+    def authenticate(self):
+        realm = "Screenly OSE {}".format(self.settings['player_name'])
+        return Response("Access denied", 401, {"WWW-Authenticate": 'Basic realm="{}"'.format(realm)})
+
+    def update_settings(self, current_password):
+        auth_backend = self.settings['auth_backend']
+        current_pass_correct = True if auth_backend == '' \
+            else self.settings.auth_backends[auth_backend].check_password(current_password)
```

Comment:
@eugenosm Please rebase your branch on top of #1162 . In my PR this line [looks different](https://github.com/Screenly/screenly-ose/pull/1162/files#diff-01a06b801eb7962fc5be15ff8f1cd899R97).